File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)

        ans = []
        sum=0
        for i in range(1,n):
            if(prices[i-1]<prices[i]):
                ans.append(prices[i]-prices[i-1])
        for j in ans:
            sum+=j
        return sum

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] prices) {
        int n= prices.length;
        List<Integer> list = new ArrayList<>();
        int sum=0;
        for(int i=1;i<n;i++){
            if(prices[i-1]<prices[i]){
                list.add(prices[i]-prices[i-1]);
            }
        }
        for(int i=0;i<list.size();i++){
            sum+=list.get(i);
        }
        return sum;
        
    }
}